### PR TITLE
[Updater] Update failures now produce human readable errors

### PR DESCRIPTION
**Pull request by Helios747**
I see people in the discord asking for help with the updater, when it's often just GitHub connectivity issues or rate limiting. I myself had gotten confused by this early on.

This commit explains the common mode of failure and recommends waiting.

### CODE REVIEW

File: XIVLauncher/Updates.cs [hunk 1]
```diff
@@ -19,23 +20,28 @@ public async Task Run(bool downloadPrerelease = false)
...
 #if !XL_NOAUTOUPDATE
-                else
-                    OnUpdateCheckFinished?.Invoke(this, null);
+                    else
+                        OnUpdateCheckFinished?.Invoke(this, null);
 #endif
-
+                }
+            }
+            catch (Exception)
```
Comment (goaaats) [inline XIVLauncher/Updates.cs:41]: You should log the Exception with Log.Error(ex, "Update failed")

File: XIVLauncher/Updates.cs [hunk 2]
```diff
@@ -19,23 +20,28 @@ public async Task Run(bool downloadPrerelease = false)
...
+            {
+                MessageBox.Show("Failed to check for updates. This may be caused by GitHub connectivity issues. Wait a few minutes and try again.", "Update failure", MessageBoxButtons.OK, MessageBoxIcon.Error);
```
Comment (goaaats) [inline XIVLauncher/Updates.cs:43]: This needs to be localized(See Loc.Localize) and please adjust the wording to include a reference to the Discord, e.g. "If this doesn't work after several minutes, please report the error on our Discord server."

**Review by goaaats — CHANGES_REQUESTED** [review]

File: XIVLauncher/Updates.cs (hunk 1 above)
Comment (Helios747) [reply XIVLauncher/Updates.cs]: Done.

File: XIVLauncher/Updates.cs (hunk 2 above)
Comment (Helios747) [reply XIVLauncher/Updates.cs]: Done.

**Pull request merged**